frontend/openchat-client/src/openchat.ts

```diff
@@ -2691,36 +2703,50 @@ export class OpenChat extends OpenChatAgentWorker {
         };
         const event = { event: msg, index: nextEventIndex, timestamp: BigInt(Date.now()) };
 
+        const canRetry = this.canRetryMessage(msg.content);
+
         this.sendRequest({
             kind: "sendMessage",
-            chatType: chat.kind,
             chatId,
             user: this.user,
             mentioned: [],
             event,
+            threadRootMessageIndex,
+            rulesAccepted,
+            communityRulesAccepted
         })
             .then(([resp, msg]) => {
                 if (resp.kind === "success") {
                     this.onSendMessageSuccess(chatId, resp, msg, undefined);
```

Comment:
should be passing threadRootMessageIndex here if we want this to work in threads too.